oodeel/datasets/torch_data_handler.py

```diff
@@ -0,0 +1,483 @@
+                the dataset. Defaults to False.
+
+        Return:
+            DictDataset: Mapped dataset
+        """
+        dataset = self if inplace else copy.deepcopy(self)
+        dataset.map_fns.append(map_fn)
+        return dataset
+
+    def filter(self, filter_fn: Callable, inplace: bool = False):
+        """Filter the dataset
+
+        Args:
+            filter_fn (Callable): filter function f: dict -> bool
+            inplace (bool): if False, applies the filtering on a copied version of\
+                the dataset. Defaults to False.
+
+        Returns:
+            DictDataset: Filtered dataset
+        """
+        indices = [
+            i
+            for i in tqdm(range(len(self)), desc="Filtering the dataset...")
+            if filter_fn(self[i])
+        ]
+        dataset = self if inplace else copy.deepcopy(self)
+        dataset._dataset = Subset(self._dataset, indices)
+        return dataset
+
+    def concatenate(self, other_dataset: Dataset, inplace: bool = False):
+        """Concatenate with another dataset
+
+        !!! warning
+            The map functions that will be retained will be those of this dataset
+
+        Args:
+            other_dataset (Dataset): Dataset to concatenate with
+            inplace (bool): if False, applies the filtering on a copied version of\
+                the dataset. Defaults to False.
+
+        Returns:
+            DictDataset: Concatenated dataset
+        """
+        dataset = self if inplace else copy.deepcopy(self)
+        dataset._dataset = ConcatDataset([self._dataset, other_dataset])
+        return dataset
+
+    def __len__(self):
+        return len(self._dataset)
+
+
+class TorchDataHandler(DataHandler):
+    """
+    Class to manage torch DictDataset. The aim is to provide a simple interface
+    for working with torch datasets and manage them without having to use
+    torch syntax.
+    """
+
+    def default_target_transform(y):
+        return torch.LongTensor([y] if isinstance(y, (float, int)) else y)
+
+    DEFAULT_TRANSFORM = torchvision.transforms.ToTensor()
+    DEFAULT_TARGET_TRANSFORM = default_target_transform
+
+    @staticmethod
+    def load_dataset(dataset_id: Any, load_kwargs: dict = {}):
+        """Load dataset from different manners
+
+        Args:
+            dataset_id (Any): dataset identification
+            load_kwargs (dict, optional): Additional loading kwargs. Defaults to {}.
+
+        Returns:
+            Any: dataset
+        """
+        if isinstance(dataset_id, str):
+            dataset = TorchDataHandler.load_torchvision_dataset(
+                dataset_id, **load_kwargs
+            )
+        elif isinstance(dataset_id, Dataset):
+            dataset = TorchDataHandler.load_custom_dataset(dataset_id)
+        elif isinstance(dataset_id, (np.ndarray, torch.Tensor, tuple, dict)):
+            dataset = TorchDataHandler.load_dataset_from_arrays(dataset_id)
+        return dataset
```

Comment:
I think this method should not be static so that it can use self.load* instead of instanciating a TorchDataLoader at each call of loading methods.